python/app/dialog.py

```diff
@@ -0,0 +1,793 @@
+            # set up a global on-click handler for
+            tab_dict["view"].doubleClicked.connect(self._on_entity_clicked)
+            # create delegate
+            tab_dict["delegate"] = DelegateClass(tab_dict["view"], self._action_manager)
+            # hook up delegate renderer with view
+            tab_dict["view"].setItemDelegate(tab_dict["delegate"])
+        
+        # set up the all fields tabs
+        self._entity_details_model = SgAllFieldsModel(self.ui.entity_info_widget)     
+        self._entity_details_model.data_updated.connect(self.ui.entity_info_widget.set_data)
+        self.ui.entity_info_widget.link_activated.connect(self._on_link_clicked)
+           
+        self._version_details_model = SgAllFieldsModel(self.ui.version_info_widget)        
+        self._version_details_model.data_updated.connect(self.ui.version_info_widget.set_data)
+        self.ui.version_info_widget.link_activated.connect(self._on_link_clicked)
+        
+        self._publish_details_model = SgAllFieldsModel(self.ui.publish_info_widget)        
+        self._publish_details_model.data_updated.connect(self.ui.publish_info_widget.set_data)
+        self.ui.publish_info_widget.link_activated.connect(self._on_link_clicked)
+        
+        # kick off
+        self._on_home_clicked()
+
+
+    def closeEvent(self, event):
+        """
+        Executed when the main dialog is closed.
+        All worker threads and other things which need a proper shutdown
+        need to be called here.
+        """        
+        
+        self._app.log_debug("CloseEvent Received. Begin shutting down UI.")
+
+        # register a shutdown overlay
+        splash_pix = QtGui.QPixmap(":/tk_multi_infopanel/bye_for_now.png")
+        self._overlay.show_message_pixmap(splash_pix)
+        QtCore.QCoreApplication.processEvents()
+
+        try:
+            
+            # clear the selection in the main views. 
+            # this is to avoid re-triggering selection
+            # as items are being removed in the models
+            #
+            # TODO: might have to clear selection models here
+            
+            CachedShotgunSchema.unregister_data_retriever(self._sg_data_retriever)
+            
+            self._sg_data_retriever.work_completed.disconnect()
+            self._sg_data_retriever.work_failure.disconnect()
+            self._sg_data_retriever.stop()
+            
+            # shut down main details model
+            self._details_model.destroy()
+            self._current_user_model.destroy()
+            
+            # and the all fields model
+            self._entity_details_model.destroy()
+            self._version_details_model.destroy()
+            self._publish_details_model.destroy()
+            
+            # gracefully close all tab model connections
+            for tab_dict in self._detail_tabs.values():
+                tab_dict["model"].destroy()            
+
+        except:
```

Comment:
As far as I understand, just `except:` is equivalent to `except Exception` - e.g. a catchall. 
Normally, it's useful to have the actual object, so we often write `except Exception, e:` to then use `e` later on, but in this case we are just logging the callstack and therefore don't need to pull out the exception object. Or is this bad practice? (apart from the pokemon approach?)